lpeval.pas

```diff
@@ -426,19 +426,41 @@ procedure _LapeWriteLn(const Params: PParamArray); {$IFDEF Lape_CDECL}cdecl;{$EN
 
 {$IFDEF Lape_NativeKeyword}
 procedure _LapeUNatify(const Params: PParamArray; const Result: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF}
+
   function getByCodePos(Compiler: TLapeCompiler; CodePos: TCodePos): TLapeGlobalVar;
   var
-    DeclArr: TLapeDeclArray;
-    I, H: UInt32;
+    DeclArr, DeclArr2: TLapeDeclArray;
+    Decl: TLapeGlobalVar;
+    I, H, J: UInt32;
   begin
+    Result := nil;
     DeclArr := Compiler.GlobalDeclarations.getByClass(TLapeGlobalVar, bTrue);
 
     H := High(DeclArr);
     for I := 0 to H do
-      if ((DeclArr[I] <> nil) and ((DeclArr[I]as TLapeGlobalVar).Ptr <> nil)) then
-        if (TCodePos((DeclArr[I] as TLapeGlobalVar).Ptr^) = CodePos) then
-          Exit(DeclArr[I] as TLapeGlobalVar);
+    begin
+      Decl := TLapeGlobalVar(DeclArr[I]);
+
+      if (Decl <> nil) and (Decl.Ptr <> nil) then
+        if (TCodePos(Decl.Ptr^) = CodePos) then
+          Exit(Decl);
+
+      if (Decl.VarType <> nil) and (Decl.VarType.DeclarationList <> nil) and

```

Comment:
Seems like you have forgot to check if  `Decl` is `nil` at this point? Should it not be:
>> if (Decl <> nil) and (Decl.VarType <> nil) and (Decl.VarType.Declar..